contracts/src/token/common/erc2981.rs

```
@@ -0,0 +1,514 @@
+        let mut royalty_fraction = &royalty_info.royalty_fraction;
+
+        if royalty_receiver.is_zero() {
+            royalty_receiver = &self.default_royalty_info.receiver;
+            royalty_fraction = &self.default_royalty_info.royalty_fraction;
+        }
+
+        let royalty_amount = (sale_price * U256::from(royalty_fraction.get()))
+            .wrapping_div(U256::from(self._fee_denominator()));
+
+        (royalty_receiver.get(), royalty_amount)
+    }
+}
+
+impl IErc165 for Erc2981 {
+    fn supports_interface(interface_id: FixedBytes<4>) -> bool {
+        <Self as IErc2981>::INTERFACE_ID == u32::from_be_bytes(*interface_id)
+    }
+}
+
+impl Erc2981 {
+    /// Function to change the denominator with which to interpret the fee set
+    /// in _setTokenRoyalty and _setDefaultRoyalty as a fraction of the sale
+    /// price  
+    /// Defaults to 10000 so fees are expressed in basis points, but
+    /// may be customized by an override.
+    ///
+    /// # Arguments
+    ///
+    /// * `&self` - Read access to the contract's state.
+    pub fn _fee_denominator(&self) -> U96 {
+        U96::from(10000)
+    }
+
+    /// Function to set the royalty information that all ids in this contract
+    /// will default to.
+    ///
+    /// # Arguments
+    ///
+    /// * `&mut self` - Write access to the contract's state.
+    /// * `receiver` - Address to receive the royalty.
+    /// * `fee_numerator` - Fraction of royalty to be given to receiver
+    ///
+    /// # Errors
+    ///
+    /// If `fee_numerator` > denominator, then the error
+    /// [`Error::InvalidDefaultRoyalty`] is returned.
+    ///
+    /// If receiver is the zero address, then the error
+    /// [`Error::InvalidDefaultRoyaltyReceiver`] is returned.
+    pub fn _set_default_royalty(
+        &mut self,
+        receiver: Address,
+        fee_numerator: U96,
+    ) -> Result<(), Error> {
+        let denominator: U256 = U256::from(self._fee_denominator());
```

You don't need to make this conversion to compare it with `fee_numerator`, you can do it once you need to return an error